llmfoundry/utils/builders.py

```diff
@@ -374,16 +382,16 @@ def _extract_param_groups(
     return model.parameters()
 
 
-def build_optimizer(
-        model: torch.nn.Module,
-        name: str,
-        optimizer_config: Optional[Dict[str, Any]] = None) -> Optimizer:
+def build_optimizer(model: torch.nn.Module, name: str,
+                    optimizer_config: Dict[str, Any]) -> Optimizer:
+
+    for k, v in optimizer_config.items():
+        if isinstance(v, DictConfig):
+            optimizer_config[k] = om.to_container(v, resolve=True)
```

Comment:
same, can't we just call to container when we call resolve so that everything is raw python objects after the initial config parsing?